python/cudf/cudf/core/column/string.py

```diff
@@ -4840,7 +4839,7 @@ def _as_datetime_or_timedelta_column(self, dtype, format):
             return cudf.core.column.as_column([], dtype=dtype)
 
         # Check for None strings
-        if (self == "None").any():
+        if (self == "None").fillna(False).any():
```

Comment:
Wouldn't the `any` filter out any `null` values anyway?

Reply:
It does filter them out. We just don't want situations like in the other comment, where one side is null, the other side is valid, and result of the comparison is then `null` and the `any()` returns the wrong answer.

Reply:
Oh wait, ignore what I just said - double checking this.

Reply:
Looks like there's a bug somewhere in the `any` reduction codepath that makes this happen:

```
>>> col = cudf.Series([None], dtype='bool')._column
>>> libcudf.reduce.reduce('any', col, 'bool')
nan
```
Which basically produces `True` incorrectly. Following up

Reply:
Found a couple of little issues when following this up. First, I tried just changing this to be `if "None" in self:`. In doing that, I noticed the following (probably) bug:

```
x = cudf.Series([None, None, None, None, None], dtype='object')._column
"None" in x
# True
```
here `x` is a series of all `<NA>`, so this should probably return `False`. OTOH, currently an `any` reduction over a fully null column returns `True`:

```
x = cudf.Series([None, None, None, None, None], dtype='object')._column
print(x.any())
# True
```

Short of fixing one or both of those bugs as part of this PR, are there any other obvious options for how to deal with this?

Reply:
We could workaround this temporarily by having a check in any that checks if `len(col) == col.null_count` for example.

Is the string issue the same or unrelated?

Reply:
Turned out to be pretty easy to fix this so I just did it as part of this PR - minor changes to reduce cython and a tweak to the `any()` logic in series. This also fixed the string issue it seems.